Document multiline matching with patterns (#198)
json-schema-org/understanding-json-schema: source/reference/regular_expressions.rst

- ``.``: Matches any character except line break characters. (Be aware that what
  constitutes a line break character is somewhat dependent on your platform and
  language environment, but in practice this rarely matters). To include

>   newlines use the ``(.|\n\r)*`` pattern which avoids the use of regex flags/modifiers

Reviewer: This should be `\r\n` not `\n\r`, right? And really it should be `\r?\n` to match UNIX style line endings as well as windows style.

Reviewer 2: Is `[\s\S]` not supported by all regex engines?

>   language environment, but in practice this rarely matters). To include
>   newlines use ``(.|\n\r)*`` pattern which avoids use of regex flags/modifiers
>   and have a very good support across regex libraries.

Reviewer 3: ```suggestion
  language environment, but in practice this rarely matters). To include
  newlines use the ``(.|\n\r)*`` pattern which avoids the use of regex flags/modifiers
  and has good support across regex libraries.
```

- ``{x}?``, ``{x,y}?``, ``{x,}?``: Lazy versions of the above
  expressions.

- Use only standard escapes like ``\n``, ``\r``, ``\t`` and keep in mind that

>   you also need to do JSON escaping.

Reviewer: I think this is a great addition. For the JSON string escaping part, I wonder if it would be better to put this somewhere near the top since this applies to things other than character classes as well.

   "(888)555-1212 ext. 532"
   --X
   "(800)FLOWERS"

The following example checks that the string starts with `{{` and ends with `}}`
and that it also allows multiline strings.

.. schema_example::

   {
      "type": "string",

>       "pattern": "^\\{\\{(.|[\r\n])*\\}\\}$",

Reviewer: Missing a couple escapes.

```suggestion
      "pattern": "^\\{\\{(.|[\\r\\n])*\\}\\}$",
```
